File: conference/registration-bc/web/src/main/java/com/microsoft/conference/registration/domain/SeatAssigning/Models/Attendee.cs
using System;
using System.Collections.Generic;
using ECommon.Utilities;
using ENode.Domain;

namespace Registration.SeatAssigning
{
    public class Attendee : ValueObject<Attendee>
    {
        public string FirstName { get; private set; }
        public string LastName { get; private set; }
        public string Email { get; private set; }

        public Attendee() { }
        public Attendee(string firstName, string lastName, string email)
        {
            Ensure.NotNull(firstName, "firstName");
            Ensure.NotNull(lastName, "lastName");
            Ensure.NotNull(email, "email");
            FirstName = firstName;
            LastName = lastName;
            Email = email;
        }

        public override IEnumerable<object> GetAtomicValues()
        {
            yield return FirstName;
            yield return LastName;
            yield return Email;
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/seatassigning/Models/Attendee.java
package com.microsoft.conference.registration.domain.seatassigning.Models;

import org.enodeframework.common.utilities.Ensure;

public class Attendee {
    public String firstName;
    public String lastName;
    public String email;

    public Attendee() {
    }

    public Attendee(String firstName, String lastName, String email) {
        Ensure.notNull(firstName, "firstName");
        Ensure.notNull(lastName, "lastName");
        Ensure.notNull(email, "email");
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
    }
}
